extensions/_proxy/heroku.py

```diff
@@ -0,0 +1,59 @@
+# -*- coding: utf-8 -*-
+'''
+Proxy minion interface module for interacting with Heroku apps.
+'''
+
+# Import Python Libs
+from __future__ import absolute_import
+import logging
+
+# import Salt Libs
+from salt.utils.dictupdate import merge
+
+DETAILS = {}
+
+log = logging.getLogger(__name__)
+
+__proxyenabled__ = ['heroku']
+__virtualname__ = 'heroku'
+
+
+def __virtual__():
+    '''
+    Only return if all the modules are available
+    '''
+    log.info('heroku proxy __virtual__() called...')
+    return __virtualname__
+
+
+def init(opts):
+    log.debug('heroku proxy init() called...')
+    log.debug('Validating heroku proxy input')
+    proxy_conf = merge(opts.get('proxy', {}), __pillar__.get('proxy', {}))
+    log.trace('proxy_conf = %s', proxy_conf)
+
+    DETAILS['proxytype'] = proxy_conf['proxytype']
+    DETAILS['initialized'] = True
```

Comment:
I think this step would be a good spot to authenticate to Heroku and then cache the authenticated client object for subsequent calls. If that is currently managed in the execution module then it would be a candidate to move to a `utils` module.

Reply:
Chatted with Tobias and he mentioned that this might not be needed since we're just passing the API key as a header for each request.